### ISSUE TITLE
How to speed up `predict_parts()` ?

### ISSUE THREAD

**Issue by MJimitater** [ModelOriented/DALEX#517]
I take reference to #514 

I noticed that running `predict_parts(explain, new_observation = test[1, , drop = FALSE], type = "break_down")` takes a VERY long time.. it  took more than somewhat 30 hours to return the result. 
This makes plotting the break_down plot for single new_observations not very handy and limits the use of this powerful model explanation method.
What am I doing wrong? How can I speed up the calculation of the break_down plot?

**Comment by MJimitater** [comment]
@pbiecek hi, so the explainer object is created with the training set of 3.2 million rows and 246 features, and the model is an xgboost model that was trained with max_depth = 3 and nrounds = 1000.
The `new_observation` within `predict_parts()` is a test set sample, 1 row and 246 features. 
I hope this helps.

**Comment by hbaniecki** [comment]
You are trying to estimate the explanation on 3.2 million observations. Just change parameter `N` to a lower number, e.g. 1000, 10000..

**Comment by MJimitater** [comment]
@hbaniecki Good point! Thank you, I will limit the observations and report back!

BTW, `modelStudio()` interrupted its calculation after 27 hours with this error: 

![image](https://user-images.githubusercontent.com/26901261/173052893-7bec908d-f9d7-4ce9-823a-baf726c67041.png)

Perhaps also something due to the large 3.2 million observations, which can be solved by limiting `N`...?!

**Comment by hbaniecki** [comment]
A few things come to mind:
1. For a description of all the `N` & `B` parameters in `modelStudio()` see https://modelstudio.drwhy.ai/articles/ms-perks-features.html#more-calculations-means-more-time
2. The `N_sv` parameter can be used to limit the `data` in `iBreakDown::shap`, but not in `iBreakDown::local_attributions` (=break_down method) as it is a considerably faster method than `shap` and was never an issue to compute in finite time -- so here `DALEX` differs from `modelStudio` at the moment.
3. All in all, I would recommend reducing the number of rows in the `data` parameter of the explainer (e.g. to 10K):
   - It is very costly to use datasets of millions of observations for computing explanations (both: global and local)
   - It will reduce the time of break down / local attributions computation like in `predict_parts(..., N=1000)`
   - Please note that you are essentially injecting a whole dataset into an HTML file of modelStudio -- this can be stopped by using `eda=FALSE`

**Comment by MJimitater** [comment]
@hbaniecki Thank you kindly for your thoughts, this is super helpful!! 
Reducing the number of rows has already increased performance significantly, without losing alot of precision worth mentioning.
I think I can surely close this issue! Many thanks
